common/src/main/java/net/william278/huskchat/player/PlayerCache.java

```diff
@@ -138,6 +149,83 @@ public static void removeLocalSpy(Player player) {
         return receivers;
     }
 
+    // Load local and social spy data into maps
+    public static void loadSpy() {
+        try {
+            YamlDocument spies = YamlDocument.create(new File(dataFolder, "spies.yml"));
+
+            if (spies.contains("local")) {
+                Section local = spies.getSection("local");
+
+                for (Object name : local.getKeys()) {
+                    localSpies.put(UUID.fromString(name.toString()),
+                            SpyColor.valueOf(local.getSection(name.toString()).getString("color")));
+                }
+            }
+
+            if (spies.contains("social")) {
+                Section social = spies.getSection("social");
+
+                for (Object name : social.getKeys()) {
+                    socialSpies.put(UUID.fromString(name.toString()),
+                            SpyColor.valueOf(social.getSection(name.toString()).getString("color")));
+                }
+            }
+        } catch(IOException e) {
+            // TODO: Use logger
+            e.printStackTrace();
+        }
+    }
+
+    // Adds spy state to data file
+    public static void addSpy(String type, UUID uuid, SpyColor spyColor) {
+        try {
+            if (!type.equals("local") && !type.equals("social")) return;
+
+            YamlDocument spies = YamlDocument.create(new File(dataFolder, "spies.yml"));
+
+            if (!spies.contains(type)) {
+                spies.createSection(type);
+            }
+
+            if (!spies.getSection(type).contains(uuid.toString())) {
+                spies.getSection(type).createSection(uuid.toString());
+            }
+
+            spies.getSection(type)
+                    .getSection(uuid.toString())
+                    .set("color", spyColor.toString());
+            spies.save();
+        } catch (IOException e) {
+            // TODO: Use logger
+            e.printStackTrace();
```

Comment:
Using the logger requires an instance of the logger class that's been adapted from the platform logger, which isn't doable nicely through the static implementation here. Best to just have these static methods throw an IOException and handle with try-catch where they're called on the platform specific classes